src/main/java/org/spongepowered/common/mixin/core/block/MixinBlock.java

```diff
@@ -119,6 +128,15 @@ public void setTickRandomly(boolean tickRandomly) {
         this.needsRandomTick = tickRandomly;
     }
 
+    @Overwrite
+    public void randomTick(World worldIn, BlockPos pos, IBlockState state, Random random) {
+        BlockRandomTickEvent event = SpongeEventFactory.createBlockRandomTick(Sponge.getGame(), null, new Location((Extent)worldIn, VecHelper.toVector(pos)));
```

Comment:
Make it `cancellable` in the `@Inject`, a `CallbackInfo` object in the parameter can then be used to return by calling `cancel()` on it.